shell/components/cards/RcItemCard.vue

```diff
@@ -0,0 +1,391 @@
+      <template v-if="variant !== 'small'">
+        <div>
+          <slot name="item-card-image">
+            <div
+              v-if="image"
+              :class="['item-card-image', variant]"
+              data-testid="item-card-image"
+            >
+              <LazyImage
+                :src="image.src"
+                :alt="labelText(image?.alt)"
+                :style="{'width': '40px', 'height': '40px', 'object-fit': 'contain'}"
+              />
+            </div>
+          </slot>
+          <slot name="item-card-pill">
+            <div
+              v-if="pill"
+              v-clean-tooltip="labelText(pill.tooltip)"
+              class="item-card-pill"
+              data-testid="item-card-pill"
+            >
+              {{ labelText(pill.label) }}
+            </div>
+          </slot>
+        </div>
+      </template>
+
+      <div :class="['item-card-body-details', variant]">
+        <div :class="['item-card-header', variant]">
+          <div class="item-card-header-left">
+            <template v-if="variant === 'small'">
+              <slot name="item-card-image">
+                <div
+                  v-if="image"
+                  :class="['item-card-image', variant]"
+                  data-testid="item-card-image"
+                >
+                  <LazyImage
+                    :src="image.src"
+                    :alt="labelText(image?.alt)"
+                    :style="{'width': '24px', 'height': '24px', 'object-fit': 'contain'}"
+                  />
+                </div>
+              </slot>
+            </template>
+            <slot name="item-card-header-title">
+              <h3
+                v-if="header.title"
+                v-clean-tooltip="labelText(header.title)"
+                :class="['item-card-header-title', variant]"
+                data-testid="item-card-header-title"
+              >
+                {{ labelText(header.title) }}
+              </h3>
+            </slot>
+          </div>
+          <div class="item-card-header-right">
+            <div
+              v-if="header.statuses?.length"
+              class="item-card-header-statuses"
+            >
+              <div
+                v-for="(status, i) in header.statuses"
+                :key="i"
+                class="item-card-header-statuses-status"
+                :aria-label="labelText(status.tooltip) || t('itemCard.ariaLabel.status')"
+                data-testid="item-card-header-statuses-status"
+              >
+                <i
+                  v-clean-tooltip="labelText(status.tooltip)"
+                  :class="['icon', status.icon, status.color]"
+                  :style="{color: status.customColor}"
+                  :data-testid="`item-card-header-status-${i}`"
+                />
+              </div>
+            </div>
+
+            <template v-if="$slots['item-card-actions']">
+              <div class="item-card-header-action-menu no-card-click">
```

Comment:
What is the `no-card-click` class for?

Reply:
I see, it's to support some functionality for the Card component in #14330 and is unrelated to this PR.